libcxx/src/filesystem/operations.cpp

```diff
@@ -82,6 +82,8 @@ path __canonical(path const& orig_p, error_code* ec) {
   path cwd;
   ErrorHandler<path> err("canonical", ec, &orig_p, &cwd);
 
+  if (orig_p.empty())
+    return err.report(capture_errno());
```

Comment:
I dislike this solution, does not fix the real issue, this part should be fixed in `__do_absolute`. See https://godbolt.org/z/GrvW1KonT

Can you fix `absolute` and `weakly_canonical` too?

Reply:
Just a heads up.

By moving that test (a slightly modified version) to `__do_absolute` as you suggest, I can get `absolute`, `canonical` and `weakly_canonical` to work as expected. However it appears that both `proximate` and `relative` are dependent on `weakly_canonical` and are both in their tests depending on the `""` becomes `current_dir` behavior.

Modifying your example, I can highlight some more discrepancies : https://godbolt.org/z/14h8Pv7vc

I believe I can iron those out, but it will take me a bit more time and extend the scope of this PR. Is that acceptable?

Reply:
Since they are all related to the same root cause I don't mind to have these changes in one PR.

Take the time you need.